content/2021-06-11-make-your-life-easier.md

```diff
@@ -0,0 +1,156 @@
++++
+title = "Make your life easier"
+template = "page.html"
+date = 2021-06-11T22:00:00Z
+
+[taxonomies]
+tags = ["Tech"]
++++
+
+If you write software for long enough you begin to develop instincts about the
+relative difficulty of different problems and some of those solidify into
+rules of thumb. Like all such things they are not really _rules_ but patterns
+of thinking that you can choose to deviate from when you have good reason to
+do so. The following are some that I have come across recently. I can't claim
+that any of them are original ideas, but they are some of the things that
+guide my work on a daily basis.
+
+# Choose simpler graph data structures
+
+Many problems are graph problems of one sort or another and there's a
```

Comment:
Who is your intended audience for this post? I think you may need to define graph before listing examples.

With your intro, I was all bought in and ready to catch some knowledge, and "learning about graph problems" seemed like a big, scary step.

Reply:
Like... when do I use a graph structure? Is my database part of a graph structure?

Reply:
I added some examples. It's probably not enough, but since I don't have a well-defined intended audience I'm not really sure where to go further than that.